src/metabase/middleware/security.clj

```diff
@@ -1,12 +1,25 @@
 (ns metabase.middleware.security
   "Ring middleware for adding security-related headers to API responses."
-  (:require [clojure.string :as str]
+  (:require [clojure.java.io :as io]
+            [clojure.string :as str]
             [metabase.config :as config]
             [metabase.middleware.util :as middleware.u]
             [metabase.models.setting :refer [defsetting]]
             [metabase.util
              [date :as du]
-             [i18n :as ui18n :refer [tru]]]))
+             [i18n :as ui18n :refer [tru]]]
+            [ring.util.codec :refer [base64-encode]])
+  (:import java.security.MessageDigest))
+
+(defn- file-hash [resource-filename]
+  (base64-encode
+    (.digest (doto (java.security.MessageDigest/getInstance "SHA-256")
+               (.update (.getBytes (slurp (io/resource resource-filename))))))))
+
+(def ^:private ^:const index-bootstrap-js-hash (file-hash "frontend_client/inline_js/index_bootstrap.js"))
+(def ^:private ^:const index-ganalytics-js-hash (file-hash "frontend_client/inline_js/index_ganalytics.js"))
+(def ^:private ^:const index-webfontconfig-js-hash (file-hash "frontend_client/inline_js/index_webfontconfig.js"))
+(def ^:private ^:const init-js-hash (file-hash "frontend_client/inline_js/init.js"))
```

Comment:
I would combine this into a single variable like this:

```clj
(defonce ^:private ^:const inline-js-hashes
  (let [file-hash (fn [resource-filename]
                    (base64-encode
                     (.digest (doto (java.security.MessageDigest/getInstance "SHA-256")
                                (.update (.getBytes (slurp (io/resource resource-filename))))))))]
    (mapv file-hash ["frontend_client/inline_js/index_bootstrap.js"
                     ;; put comments here instead
                     "frontend_client/inline_js/index_ganalytics.js"
                     "frontend_client/inline_js/index_webfontconfig.js"
                     "frontend_client/inline_js/init.js"])))
```

Since the `file-hash` function isn't needed outside of those 4 calculations if you inline inside the body of the `^:const` variable then the compiler should be able to toss it after so the unused fn won't go in the final product.

Also when people are working in the REPL (possibly) recalculating those 4 hashes every time the namespace is reloaded is a little annoying so it's better to just make the `def` a `defonce` instead.

Reply:
Clever! Done.